configd/src/apps/su/main.cpp

```diff
@@ -36,7 +40,11 @@ int main(int argc, char** argv)
         return 1;
     }
     size_t listsize = 1;
-    gid_t grouplist[1] = { g };
+    if (ggl > 0) {
```

Comment:
You might want to use`ggl  >= 0` here, since getgrouplist returns 0 when successful on darwin.